src/shared/components/gene-search-panel/GeneSearchPanel.tsx

```diff
@@ -0,0 +1,296 @@
+    </div>
+  );
+
+  if (searchMatches.length === 1) {
+    return (
+      <tr>
+        <td>{speciesElement}</td>
+        <td>
+          <GeneMatch match={searchMatches[0]} species={speciesInfo} />
+        </td>
+      </tr>
+    );
+  } else {
+    return (
+      <>
+        {searchMatches.map((match, index) => (
+          <tr key={index}>
+            {index === 0 && (
+              <td rowSpan={searchMatches.length}>{speciesElement}</td>
+            )}
+            <td>
+              <GeneMatch match={match} species={speciesInfo} />
+            </td>
+          </tr>
+        ))}
+      </>
+    );
+  }
+};
+
+const GeneMatch = (props: { match: SearchMatch; species: CommittedItem }) => {
+  const {
+    match: { unversioned_stable_id },
+    species: { genome_tag, genome_id }
+  } = props;
+  const [shouldShowTooltip, setShouldShowTooltip] = useState(false);
+  const anchorRef = useRef<HTMLSpanElement>(null);
+
+  const genomeIdForUrl = genome_tag || genome_id;
+
+  const handleClick = () => {
+    setShouldShowTooltip(!shouldShowTooltip);
+  };
+
+  const urlForGenomeBrowser = urlFor.browser({
+    genomeId: genomeIdForUrl,
+    focus: buildFocusIdForUrl({ type: 'gene', objectId: unversioned_stable_id })
+  });
+
+  const urlForEntityViewer = urlFor.entityViewer({
+    genomeId: genomeIdForUrl,
+    entityId: buildFocusIdForUrl({
+      type: 'gene',
+      objectId: unversioned_stable_id
+    })
+  });
+
+  // NOTE: If this panel opens in genome browser, we should add `replaceState: true`
+  // to the genomeBrowser link config below
+  const links = {
+    genomeBrowser: {
+      url: urlForGenomeBrowser
+    },
+    entityViewer: {
+      url: urlForEntityViewer
+    }
+  };
+
+  return (
+    <>
+      <span className={styles.geneMatch} ref={anchorRef} onClick={handleClick}>
+        {props.match.stable_id}
+      </span>
+      {shouldShowTooltip && anchorRef.current && (
+        <PointerBox
+          anchor={anchorRef.current}
+          classNames={{ box: styles.tooltip, pointer: styles.tooltipTip }}
+          position={PointerBoxPosition.RIGHT_TOP}
+          onOutsideClick={handleClick}
+        >
+          <ViewInApp links={links} />
+        </PointerBox>
+      )}
+    </>
+  );
+};
+
+type SearchMatchesWithSpecies = {
+  speciesInfo: CommittedItem;
+  searchMatches: SearchMatch[];
+};
+
+const getGroupedSearchMatches = (
+  speciesList: CommittedItem[],
+  searchMatches: SearchMatch[]
+) => {
+  const searchMatchesMap = new Map<string, SearchMatchesWithSpecies>(); // will preserve insertion order, i.e. the order of committed species
+  speciesList.forEach((species) => {
+    searchMatchesMap.set(species.genome_id, {
+      speciesInfo: species,
+      searchMatches: []
+    });
+  });
+
+  searchMatches.forEach((match) => {
+    const matches = searchMatchesMap.get(match.genome_id)?.searchMatches;
+    if (matches) {
+      matches.push(match);
```

Comment:
Where is `matches` used?

Reply:
Good point. I can't remember what this was for. Perhaps it's remnant from before I used speciesList.forEach above. I'll remove it.

Reply:
`matches` is an array that's seating in searchMatchesMap (line 279 above) 😃 On line 283 we iterate through all search matches (a flat array of search matches, not grouped by species), then, for each match, on line 284, we look up in the map (filled on lines 277-279) an entry under the same genome id that a search match has; and then we push the search match into that array. As a result we distribute the flat array of search matches into groups, each keyed by the genome id.

Reply:
OK, so `matches` is a reference?

Reply:
Yes, line numbers are now off. This is what the matches are

![image](https://user-images.githubusercontent.com/6834224/234874262-629c00d5-40a3-49ad-8121-248050afef1c.png)